src/components/surveys/survey-design/question-edit/rhs-subcontrols/Year.tsx

```diff
@@ -110,54 +66,56 @@ const Year: React.FunctionComponent<{
     if (range.max && range.min && range.max < range.min) {
       return 'RANGE'
     }
-    if (gtCurrent && !allowFuture) {
+    if (gtCurrent && allowFuture !== undefined && !allowFuture) {
```

Comment:
Handle case where `allowFuture` is undefined when a maximum year value is set for the maximum year radio group.